### DISCUSSION TITLE
dummy adder

### DISCUSSION THREAD

**Discussion by narutozxp** [YosysHQ/yosys#3760] [General]
### Feature Description

As we hope to using the adder chain in the FPGA，However， the cin signal of adder can only get from the previous adder so we hope yosys could generate a dummy adder to output the first cin signal.
Is there any way that could realize our goal?
Thanks a lot !

**Comment by gatecat** [comment]
Yes, there are several architectures where we do such things already.

For example; on Intel ALM we create a feed-in carry ALM only if the first carry input is anything other than a constant zero:
https://github.com/YosysHQ/yosys/blob/d82bae32bee63d4a521e5cb081359aa5a35213f1/techlibs/intel_alm/common/arith_alm_map.v#L36-L49

For FABulous, we create the feed-in carry LUT always:
https://github.com/YosysHQ/yosys/blob/d82bae32bee63d4a521e5cb081359aa5a35213f1/techlibs/fabulous/arith_map.v#L36-L43

**Reply by narutozxp** [reply]
oh, thanks, but it seems, in FABulous, that the adders use the input signal of lut ,rather than the output signal. However, our adders use the output of lut, so the yosys should insert a lut to supply the input signal of adder. 
Is there any doc that could help me to realize our goal?

**Reply by Ravenslofty** [reply]
The Intel ALM code above has the LUT and carry chain grouped together into a `MISTRAL_ALUT_ARITH` to achieve that.
